src/main/java/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,47 @@
+package step2;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private static final String BASIC_DELIMITER = ",|:";
+    private static final String SPECIAL_DELIMITER = "//(.)\n(.*)";
+
+    private static final String NEGATIVE_NUMBER_ERROR = "문자열에 음수가 포함되어 있습니다.";
+
+    private static final Pattern SPECIAL_PATTERN = Pattern.compile(SPECIAL_DELIMITER);
+
+    public static int splitAndSum(String numberStr) {
+        if (numberStr == null || numberStr.isBlank()) {
+            return 0;
+        }
+
+        List<String> numbers = split(numberStr);
+        checkPositive(numbers);
+
+        return numbers.stream()
+                      .mapToInt(Integer::parseInt)
+                      .sum();
+    }
+
+    private static List<String> split(String numberStr) {
+        Matcher m = SPECIAL_PATTERN.matcher(numberStr);
+        if (m.find()) {
+            String delimiter = m.group(1);
+            return List.of(m.group(2).split(delimiter));
+        }
+
+        return List.of(numberStr.split(BASIC_DELIMITER));
+    }
+
+    private static void checkPositive(List<String> numbers) {
+        for (String number : numbers) {
+            int parsedInt = Integer.parseInt(number);
+            if (parsedInt < 0) {
+                throw new IllegalArgumentException(NEGATIVE_NUMBER_ERROR + " - " + number);
+            }
+        }
+    }
```

Comment:
```suggestion
    private static int sum(List<String> numbers) {
        int result = 0;
        for (String number : numbers) {
            int parsedInt = Integer.parseInt(number);
            checkPositive(parsedInt);
            result += parsedInt;
        }
        return result;
    }
```

양수 검증을 위해서 문자열을 int로 변환하고 
합산할 때 문자열을 다시 int로 변환하는 작업을 하고 있네요.
sum 내부에서 양수인지 음수인지 한번에 체크할 수 있지 않을까요?